dawn/main/ansible/Ansible.js

```diff
@@ -177,28 +180,29 @@ class SendSocket {
 }
 
 class TCPSocket {
-  constructor(socket) {
+  constructor(socket, logger) {
     this.waitRuntimeConfirm = this.waitRuntimeConfirm.bind(this);
     this.tryUpload = this.tryUpload.bind(this);
+    this.logger = logger;
```

Comment:
Logger is always set before this part. It is initialized at the very bottom, right before any of the other classes are running.